src/main/java/org/spongepowered/api/text/action/ClickAction.java

```diff
@@ -29,35 +29,84 @@
 /**
  * Represents a {@link TextAction} that responds to clicks.
  *
- * @param <R> the type of the result of the action
+ * @param <R> The type of the result of the action
  */
-public interface ClickAction<R> extends TextAction<R> {
+public abstract class ClickAction<R> extends TextAction<R> {
```

Comment:
Every instance of this should extend this class there is no point to construct this directly